### ISSUE TITLE
Automate GitHub labeling

### ISSUE THREAD

**Issue by Borda** [PyTorchLightning/pytorch-lightning#9519]
## 🚀 Feature

Updating rules for using "help wanted" labels.
The `help wanted` label shall be used for all bugs
Issues: shall be in one of these three "states":
- triage state
- `help wanted` (we have decided to implement it)
- `good first issue` (a special case of help wanted)
- being assigned to a user/contributor

Automation/workflow:
- when assigned remove `help wanted`
- when `good first issue` remove `help wanted`
- when added `let's do it` add `help wanted`

### Motivation

Make more intuitive which labels issues are meant for community and which would be better to keep for maintainers

### Additional context

this shall contain Tempkate update as well as CI automations

**Comment by akihironitta** [comment]
This issue should be addressed before #9216.

**Comment by carmocca** [comment]
This should be easier after #10473 as it provides more fine-grained labels

**Comment by carmocca** [comment]
This can be done now.

https://github.com/pytorch/probot#auto-label-bot has auto-label functionality.
However, they don't publish the bot as an action which means we would have to maintain it and publish it as it was done for the auto-cc bot.

For this reason, I would try to find an action or app that supports what we want to do, and if nothing is good enough, go with our own.
I suggest you come up with the list of rules you would want to add first.

**Comment by carmocca** [comment]
Closing this. Low priority and prone to false positives/negatives
